.cirrus.yml

```diff
@@ -22,584 +40,532 @@ task:
-      goldctl_script: ./dev/bots/download_goldctl.sh
-      test_script:
+        CPU: 3
+      script:
+        - ./dev/bots/download_goldctl.sh
         - dart --enable-asserts ./dev/bots/test.dart
+
+    - name: framework_tests-misc-linux
+      # this includes the tests for directories in dev/
+      only_if: "changesInclude('.cirrus.yml', 'dev/**', 'packages/flutter/**', 'packages/flutter_test/**', 'packages/flutter_tools/lib/src/test/**', 'bin/internal/**') || $CIRRUS_PR == ''"
       container:
-        cpu: 4
-        memory: 12G
-    - name: tests_extras-linux
-      skip: "!changesInclude('dev/**', 'packages/flutter/**', 'packages/flutter_test/**', 'packages/flutter_tools/lib/src/test/**', 'bin/internal/**', 'dev/**') && $CIRRUS_BRANCH != 'master' && $CIRRUS_BRANCH != 'stable' && $CIRRUS_BRANCH != 'beta' && $CIRRUS_BRANCH != 'dev'"
+        cpu: 3
       env:
-        GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
-        SHARD: tests
-        SUBSHARD: extras
-      test_script:
+        CPU: 3
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
+
+    - name: tool_tests-general-linux
+      only_if: "changesInclude('.cirrus.yml', 'dev/**', 'packages/flutter_tools/**', 'bin/internal/**') || $CIRRUS_PR == ''"
       container:
+        # As of October 2019, the tool_tests-general-linux shard got faster with more CPUs up to 4
+        # CPUs, and needed at least 8G of RAM to not run out of memory.
         cpu: 4
-        memory: 12G
-    # all of the tests except the ones in test/integration and test/commands/create_test for packages/flutter_tools
-    - name: tool_tests_general-linux
-      skip: "!changesInclude('dev/**', 'packages/flutter_tools/**', 'bin/internal/**') && $CIRRUS_BRANCH != 'master' && $CIRRUS_BRANCH != 'stable' && $CIRRUS_BRANCH != 'beta' && $CIRRUS_BRANCH != 'dev'"
+        memory: 8G
       env:
-        GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
-        SHARD: tool_tests
-        SUBSHARD: general
-      test_script:
+        CPU: 4
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
+
+    - name: tool_tests-commands-linux
+      only_if: "changesInclude('.cirrus.yml', 'dev/**', 'packages/flutter_tools/**', 'bin/internal/**') || $CIRRUS_PR == ''"
       container:
-        cpu: 4
-        memory: 12G
-    - name: tool_tests_commands-linux
-      skip: "!changesInclude('dev/**', 'packages/flutter_tools/**', 'bin/internal/**') && $CIRRUS_BRANCH != 'master' && $CIRRUS_BRANCH != 'stable' && $CIRRUS_BRANCH != 'beta' && $CIRRUS_BRANCH != 'dev'"
+        # As of October 2019, the tool_tests-commands-linux shard got faster with more CPUs up to 6
+        # CPUs, and needed at least 8G of RAM to not run out of memory.
+        cpu: 6
+        memory: 8G
```

Comment:
Why not just increase this to the max we can without running into scheduling issues?  Is there a penalty for 8 CPUs with 8GB of memory for example?

Reply:
The max we can increase this to without scheduling issues is 0.1 CPU and 256M of RAM, insofar as everything above the minimum reduces the potential pool further.